src/containers/forms/LogActivityForm.jsx

```diff
@@ -28,37 +52,101 @@ class LogActivityForm extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      selectValue: '',
+      activityTypeId: '',
+      date: '',
+      description: '',
+      errors: [],
+      message: null,
     };
   }
-  onSubmit = (e) => {
-    e.preventDefault();
-  }
 
   /**
    * @memberOf LogActivityForm
    * change event handler
    * @param {Event} event - change event on select input
    */
   handleChange = (event) => {
-    this.setState({ selectValue: event.target.value });
+    this.setState({ [event.target.name]: event.target.value });
+    // if input value is not empty remove it from error list
+    if (!this.state[event.target.name]) {
+      const errors = this.state.errors.filter(error => !error.includes(event.target.name));
+      this.setState({ errors });
+    }
+  }
+
+  handleAddEvent = (event) => {
+    event.preventDefault();
+    const { activityTypeId, date, description } = this.state;
+    const activity = {
+      activityTypeId,
+      date,
+      description,
+    };
+    this.setState({
+      errors: validateFormFields(activity),
+    }, () => {
+      if (this.state.errors.length === 0) {
+        this.props.createActivity(activity);
+      }
+    });
+  }
+
+  /**
+   * @name resetState
+   * @summary resets state to clear form fields and error messages
+   */
+  resetState = () => {
+    this.setState({
+      activityTypeId: '',
+      date: '',
+      description: '',
+      errors: [],
+      message: null,
+    });
+  }
+
+  /**
+   * @name cancelModal
+   * @summary reset state and close modal
+   */
+  cancelModal = (event) => {
+    event.preventDefault();
+    this.resetState();
+    this.props.closeModal();
+  }
+
+  renderValidationError = (field, replaceWord) => {
+    if (this.state.errors.indexOf(field) >= 0) {
+      return `${capitalizeString(replaceWord || field)} is required`;
+    }
+    return '';
   }
 
   render() {
-    const { selectValue } = this.state;
+    const { selectValue, message } = this.state;
     const { categories } = this.props;
 
     return (
-      <form onSubmit={this.onSubmit}>
+      <form>
         <div className='titleForm'>Log an Activity</div>
-        <DateField />
+        <DateField
+          handleChange={this.handleChange}
+          value={this.state.date}
+        />
+        <span className='validate__errors'>
+          {this.renderValidationError('date')}
+        </span>
         <Select
-          name='fellowActivities'
+          name='activityTypeId'
           placeholder='Select Category'
           options={categories}
           title='Activity Category'
+          value={this.state.activityTypeId}
           handleChange={this.handleChange}
         />
+        <span className='validate__errors'>
+          {this.renderValidationError('activityTypeId', 'Category')}
+        </span>
         {
           selectValue === 'eef0e594-43cd-11e8-87a7-9801a7ae0329' ?
```

Comment:
What is this hardcoded UUID about?

Reply:
That was a quick fix at the time to display an optional input field when the category selected is Bootcamp interviews. My next task is to work on a bug story to fix this